mobile-app/src/atoms/friendsAtom.ts

```diff
@@ -0,0 +1,75 @@
+import { collection, onSnapshot, or, query, where } from "firebase/firestore";
+import { atom } from "jotai";
+import { atomFamily } from "jotai/utils";
+import {
+  fetchCommonHabitIds,
+  fetchFriendData,
+  fetchMutualFriends,
+} from "../firebase/api";
+import { firestore } from "../firebase/config";
+import { allUsersInfoT } from "../lib/db_types";
+import { currentUserIdAtom } from "./currentUserAtom";
+import { atomStore } from "./store";
+
+export const allFriendsDataAtom = atom<allUsersInfoT>({});
+allFriendsDataAtom.onMount = (set) => {
+  // refetch everytime users friends change
+  const currentUserId = atomStore.get(currentUserIdAtom);
+  const q = query(
+    collection(firestore, "friendships"),
+    or(
+      where("user1Id", "==", currentUserId),
+      where("user2Id", "==", currentUserId),
+    ),
+  );
+  const unsubscribeFriendlist = onSnapshot(q, () => {
+    fetchFriendData({ userId: currentUserId }).then(set);
+  });
+  return () => {
+    // close the socket when allFriendsDataAtom is unmount
+    unsubscribeFriendlist();
+  };
+};
```

Comment:
Move fetching logic to api. You can pass the setter function into the api, so that it can set the atom when new info comes in. And then you can make the api function return the unsubscribe hook so that you can get it here and call it on unmount.